rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -540,18 +567,23 @@ def _set_group_attributes(self, sm, group, request_dict):
             group.default_blueprint = sm.get(
                 models.Blueprint, request_dict['blueprint_id'])
 
-    def _set_group_deployments(self, sm, group, request_dict):
+    def _set_group_deployments(self, sm, group, request_dict, clear=True):
         deployment_ids = request_dict.get('deployment_ids')
         if deployment_ids is not None:
             deployments = [sm.get(models.Deployment, dep_id)
                            for dep_id in deployment_ids]
-            group.deployments.clear()
+            if clear:
+                group.deployments.clear()
             for dep in deployments:
                 group.deployments.append(dep)
 
         deployment_count = len(group.deployments)
         rm = get_resource_manager()
         input_overrides = request_dict.get('inputs') or []
+        if input_overrides and not group.default_blueprint:
```

Comment:
Why does the default blueprint only matter when there are input overrides?

Reply:
`input_overrides` is the only way we actually create new deployments in this, so only then we need a blueprint

Reply:
(see below for the deployment-creation loop)